ntbs-service/wwwroot/source/Components/InactivityLeaveComponent.ts

```diff
@@ -0,0 +1,66 @@
+﻿import Vue from "vue";
+
+const timerTemplate = `
+    <div class="dialog-box">
+        <section>
+            <h3 class="dialog-box-message">{{ message }}</h3>
+            <h5>You will be automatically logged out of NTBS in {{ Math.floor(timer/60) }}m {{timer%60}}s due to inactivity.</h5>
+            <div class="dialog-box-button-container">
+                <button class="nhsuk-button nhsuk-button--secondary dialog-box-button" type="button" @click="rejectHandler">Stay</button>
+                <button class="nhsuk-button nhsuk-button--secondary dialog-box-button" type="button" @click="resolveHandler">Logout</button>
+            </div>
+        </section>
+    </div>
+    `;
+
+export default Vue.extend({
+    name: 'InactivityLeaveComponent',
+    template: timerTemplate,
+    props: {
+        message: {
+            type: String,
+        },
+        resolve: {
+            type: Function,
+        },
+        reject: {
+            type: Function,
+        },
+    },
+    data: function () {
+        return {
+            timer: 120,
+            counter: false,
+            interval: null
+        }
+    },
+    mounted() {
+        this.interval = setInterval(this.countDown, 1000);
+    },
+    methods: {
+        resolveHandler() {
+            // Resolve promise within vue-accessible-modal
+            // In this implementation triggers .Confirm().then() block in FormLeaveChecker
+            this.resolve();
+        },
+        rejectHandler() {
```

Comment:
Not sure exactly how this works - would clicking outside of the box also reject the handler?
Somewhat separately, I wonder if it would at all be cleaner to always resolve due to user action, but resolve with a an appropriate value? Not sure if it would be nicer 🤔